libgphoto2/gphoto2-setting.c

```diff
@@ -179,7 +179,10 @@ load_settings (void)
 	SHGetFolderPath(NULL, CSIDL_PROFILE, NULL, 0, buf);
 	strcat (buf, "\\.gphoto");
 #else
-	snprintf (buf, sizeof(buf), "%s/.gphoto", getenv ("HOME"));
+	if (getenv("XDG_CONFIG_HOME") != NULL)
```

Comment:
Also what if `XDG_CONFIG_HOME` is set but empty? The empty string is returned here.

(same below)